tests/bip/test_seed.py

```diff
@@ -0,0 +1,121 @@
+             "94267cd146456f0aeb53459380075ea16583f3791542b3eaedce7f3c04e6bd0f"),
+        )
+        for path, ref_public_key, ref_chaincode in secp256k1_test_values:
+            pk, cc = calculate_public_key_and_chaincode(curve=CurveChoice.Secp256k1,
+                                                        path=path)
+            self.assertEqual(pk, ref_public_key)
+            self.assertEqual(cc, ref_chaincode)
+
+    def test_get_public_key_and_chaincode_nist256p1(self):
+        nist256p1_test_values: List[Tuple[str, str, str]] = (
+            ("m/44'/0'",
+             "04cc70e8a1d99dd7dfa962e22f236d1be2f033714644a4c3515c64b8b9ba63e56b95ce3053c98733b5c0cc1e376ead8fed94cc78b0b2bf1bca5e12135ea864c8b8",
+             "0aca88bf7725c7a2848f469d4d75f892f121d52307f63727968f5db1c68f06aa"),
+            ("m/44'/0'/0'/0/0",
+             "0448c284314849a6034315dd58362891b8a10e3d63b60d1409f3be4736b7dce740aaa16cd05df874f3b1dd291293b0a01dd2aa851db6c6143366ecc78a82db24ac",
+             "d8d006dc8199b09aacd77494db7803eb17ce090c9f018c1dfd80dcf44353a98a"),
+            ("m/44'/601'/1430'/20'/0'/0'",
+             "047f3ff931a503774d82fd01c8a6eab20abb45395ca76872aa65ed06828b355a285b2a73fe0dcb63efa0f75f943460e335bb01920c412a63caae17898ab9d84bd2",
+             "c8f697a463fcb0c52a5220766f7b58f503d14b2c79e37a82320e17b9bb5bc741"),
+            ("m/44'/601/1430/20/0/0",
+             "0457ba6d8ab14a9184dbcf9a3fb32d8cf1011cc68f4883955e22f51c9167424822493182a6c6a19cf88437b426ab6f96c2e557a0ff68f6c19a01ad9c2562660624",
+             "45e506e3dc52bdd0752d035bd0262f3c518bcd1a2f4e9fdaf1201f4dc26541c3"),
+        )
+        for path, ref_public_key, ref_chaincode in nist256p1_test_values:
+            pk, cc = calculate_public_key_and_chaincode(curve=CurveChoice.Nist256p1,
+                                                        path=path)
+            self.assertEqual(pk, ref_public_key)
+            self.assertEqual(cc, ref_chaincode)
+
+    def test_get_public_key_and_chaincode_ed25519slip(self):
+        ed25519slip_test_values: List[Tuple[str, str, str]] = (
+            ("m/44'/0'",
+             "00878ee3e0514e12dbbfa81a46f5bfea2e2266fef56e6e2fac64cac2d244a1b1bf",
+             "4103e5fb91a143a23db9a210e698c196d146f7e7cad678d598bacf80efe7e28d"),
+            ("m/44'/0'/0'/0'/0'",
+             "00c2ea5e68ab428901e21cb1370765b4cff3abea905d4b4b2efebf977c6202782a",
+             "b40ac81f3f3a3edc493edc95fcb2a7b1735aa2a5a706f0ee4511c3d0f5dfaf06"),
+            ("m/44'/601'/1430'/20'/0'/0'",
+             "0059e258339ae430ded97912ed077038edf189ceb8f7d022741a9000950b131f30",
+             "a6e90bed1450e77a6f3933b075fa23be315e676ccd5eddd778a62e3d669ab2d6"),
+        )
+        for path, ref_public_key, ref_chaincode in ed25519slip_test_values:
+            pk, cc = calculate_public_key_and_chaincode(curve=CurveChoice.Ed25519Slip,
```

Comment:
I feel like these 5 tests are the same, except for the input curve and the function result. Couldn't they be factorized / parametrized?

Reply:
Done, as much as I could (not that much)